impeller/core/host_buffer.cc

```diff
@@ -137,10 +143,13 @@ std::pair<uint8_t*, Range> HostBuffer::HostBufferState::Emplace(
 }
 
 void HostBuffer::HostBufferState::Reset() {
-  generation += 1;
-  device_buffer = nullptr;
-  bool did_truncate = Truncate(0);
-  FML_CHECK(did_truncate);
+  offset = 0u;
+  current_buffer = 0u;
+  device_buffers.clear();
```

Comment:
Any plans for reusing the `DeviceBuffer`s across frames? One easy solution would be to rotate through `MaxFramesInFlight` vectors IMO.

Reply:
Yeah that seems simple enough. Added this logic plus some trimming of buffers if they go unused.